assets/src/stories-editor/blocks/amp-story-page-attachment/edit.js

```diff
@@ -81,42 +59,39 @@ class PageAttachmentEdit extends Component {
 			color: textColor.color,
 			backgroundColor: appliedBackgroundColor,
 		};
+
 		return {
 			style: attachmentStyle,
 			attachmentClass,
 		};
-	}
+	};
 
-	render() {
-		const {
-			attributes,
-			setAttributes,
-		} = this.props;
+	useEffect( () => {
+		const { style, attachmentClass } = getWrapperAttributes();
+		const newAttributes = { wrapperStyle: style };
+		newAttributes.attachmentClass = attachmentClass;
+		setAttributes( newAttributes );
+	}, [ backgroundColor, customBackgroundColor, textColor ] );
```

Comment:
You need everything used by `getWrapperAttributes()` in the guard as well - it's complex, but you can end up with a situation where the `useEffect` references and "older" `getWrapperAttributes`, that uses an "older" `colors` variable and thus is out of sync with the component as a whole. Using multiple hooks, especially `useEffect` can give some really weird side-effects when some of the are reused and some aren't and they reference different callbacks, etc.

Either add `colors` to this guard or use a `useRef` for `colors` to make sure that you always access the most recent value.

Looking at it more closely, why not inline the functionality of `getWrapperAttributes` in the effect directly? It's only used here as far as I can see?

Reply:
Fixed in 3ccbe2bec262efacbc5652911a87112b4f0555c6